### PR TITLE
Resolves mesonPath to program absolute path

### PR DESCRIPTION
**Pull request by xclaesse**
ProcessExecution() does not lookup into PATH.
    
Fixes: #181

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -220,3 +222,7 @@ export function whenFileExists(ctx: vscode.ExtensionContext, file: string, liste
     listener();
   }
 }
+
+export function mesonProgram(): string {
+  return which.sync(extensionConfiguration("mesonPath"));
```
Comment (tristan957) [inline src/utils.ts:227]: Can you make this a variable instead? Otherwise, we are calling this a lot.

**Comment by tristan957** [comment]
@ZZH-Finzlize can you also try this PR?

**Comment by ZZH-Finzlize** [comment]
@tristan957 Yes, I tried this PR, it can solve #203 as well, and I noticed that this PR will call meson program by absolute path

Test result:
image

**Comment by lukester1975** [comment]
Hmm. `ProcessExecution` does look in the path though. Nothing would have ever worked otherwise (the default for `mesonPath` is just `meson`, after all)...

In fact, trying direnv and before any changes related to #181, it picks up meson from `.envrc` just fine for me. Maybe the initial issue from #181 was just casued by a misconfiguration?

**Comment by paveloom** [comment]
There is hardly anything to misconfigure. I checked and even commented on https://github.com/mesonbuild/vscode-meson/pull/186.

3e9cd2317d3be9a0fa3a77cb801c8b89da3ddd7a:

https://github.com/mesonbuild/vscode-meson/assets/49961859/aaa32bb7-1e46-4a22-84cb-086d92a91e20

5a103c462f5a9d4e17bd9184e5b4fcfff3e1495c:

https://github.com/mesonbuild/vscode-meson/assets/49961859/499a5ffa-cc13-4f68-b83b-28f44bf1f64d

**Comment by lukester1975** [comment]
I guess I may have run vscode from the already direnved environment such that the PATH was correct, without really needing the direnv plugin to do anything. Maybe `ProcessExecution` only sees the pre-direnved environment's PATH...

Anyway no matter since this latest iteration does the job, was just more curious about what was going on!

**Pull request merged**